libs/vscode/messaging/src/lib/pipe.ts

```diff
@@ -0,0 +1,49 @@
+import { mkdirSync, unlinkSync } from 'fs';
+import { platform, tmpdir } from 'os';
+import { join, resolve } from 'path';
+import { importNxPackagePath } from '@nx-console/shared-npm';
+
+const DAEMON_DIR_FOR_CURRENT_WORKSPACE = join('.nx', 'workspace-data', 'd');
+
+async function socketDirName(workspaceRoot: string): Promise<string> {
+  const { hashArray } = await importNxPackagePath<
+    typeof import('nx/src/native')
+  >(workspaceRoot, 'src/native');
+  const unique = hashArray([workspaceRoot, 'nx-console']);
```

Comment:
do you know what version compat this implies? Has hashArray been around since nx 15? Otherwise we'll have to raise the supported range of versions

Reply:
They wouldn't even be able to use this feature if they're on Nx 15. If this fails to import, then we fall back to the const above, which would use `.nx/workspace-data/d` as the docket path.